Don't skip empty transfers. In isochronous transfers, make sure

zagrodzki opened the pull request:
there is enough packets to fit the entire buffer (I think the previous
implementation would have come up short by 1 packet if the buffer
length was not an exact multiple of packet size).

Addresses https://github.com/google/gousb/issues/71

QuLogic commented:
Strange, the existing iso test also has buffer length != multiple of packet size, doesn't it? Or is the test not quite checking what it should?

zagrodzki commented:
Yeah, that test was confusing, also the implementation of the fake libusb allowed transfers that USB wouldn't allow.

Looking at the details, the buffer attached to the transfer has the full length of the passed buffer, but the transfer will return at most numIsoPackets * isoPktSize. I've updated the fake libusb to honor the packet sizes, which uncovered a few other places where the test was incorrectly assuming the transfer would be larger than actually allowed. 

That implementation change was wrong, I think - it would have allowed a transfer that is bigger than the underlying buffer which would receive the contents of the sum of all iso transfers. Reverted that part.

stapelberg commented:
Looks good, but please clean up the git commit history before merging :)

zagrodzki commented:
Squash merge will take care of that :)
